src/components/connect.js

```diff
@@ -192,7 +196,14 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
         this._unmounted = true
       }
 
+      shouldComponentUpdate(nextProps, nextState) {
+        return !options.pure ||
+          this.state.data != nextState.data ||
+          (dependsOnProps ? (this._propsDidChange || this.props.children != nextProps.children) : !shallowEqual(this.props, nextProps))
```

Comment:
this would be simpler if I were to shallow compare the props again, is the complexity worth the performance gain from not calling `shallowEqual` again for the props?
(if the reason for `shouldComponentUpdate` being called is the props having changed, `componentWillReceiveProps` will have been called just before)